python/utils/terra_utils/terra_util.py

```diff
@@ -695,17 +695,29 @@ def update_workspace_attributes(self, attributes: list[dict]) -> None:
             content_type="application/json"
         )
 
-    def leave_workspace(self, workspace_id: Optional[str] = None) -> None:
+    def leave_workspace(self, workspace_id: Optional[str] = None, ignore_direct_access_error: bool = False) -> None:
         """
         Leave a workspace. If workspace ID not supplied will look it up
 
         Args:
             workspace_id (Optional[str], optional): The workspace ID. Defaults to None.
+            ignore_direct_access_error (Optional[bool], optional): Whether to ignore direct access errors.
+             Defaults to False.
         """
         if not workspace_id:
             workspace_info = self.get_workspace_info()
             workspace_id = workspace_info['workspace']['workspaceId']
-        self.request_util.run_request(
+        accepted_return_code = [403] if ignore_direct_access_error else []
+        res = self.request_util.run_request(
             uri=f"{SAM_LINK}/resources/v2/workspace/{workspace_id}/leave",
-            method=DELETE
+            method=DELETE,
+            accept_return_codes=accepted_return_code
         )
+
+        if (res.status_code == 403
+                and res.json()["message"] == "You can only leave a resource that you have direct access to."
+                and ignore_direct_access_error):
```

Comment:
Doesn't really matter, but it can't get here unless `ignore_direct_access_error` is being used because it will fail in `run_request` if it gest 403 and you didn't tell it to ignore that return code.

Reply:
good point! wrapped in try/except to handle that